python/ray/tune/execution/tune_controller.py

```diff
@@ -1896,8 +1899,6 @@ def _try_recover(self, trial: Trial, exc: Union[TuneError, RayTaskError]):
         # Resetting this, in case that the trial is in saving status when it crashes.
         if trial.is_saving:
             trial.temporary_state.saving_to = None
-        if trial.is_restoring and exc:
-            exc = _TuneRestoreError(exc)
         self._schedule_trial_stop(trial, exception=exc)
```

Comment:
Do we need to call `trial.handle_error(exception)` before this one?

Reply:
`try_recover` only gets called in `process_trial_failure`, which already calls `handle_error`.